packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -379,24 +375,39 @@ async function sendConversionResult(
     }
   );
 
-  log(`Sending result info to queue`);
+  log(`Sending result info to queues`);
   const queuePayload = JSON.stringify({
     s3BucketName: conversionResultBucketName,
     s3FileName: fileName,
   });
 
-  const sendParams = {
-    MessageBody: queuePayload,
-    QueueUrl: conversionResultQueueURL,
-    MessageAttributes: {
-      ...SQSUtils.singleAttributeToSend("cxId", cxId),
-      ...SQSUtils.singleAttributeToSend("patientId", patientId),
-      ...(jobStartedAt ? SQSUtils.singleAttributeToSend("jobStartedAt", jobStartedAt) : {}),
-      ...(jobId ? SQSUtils.singleAttributeToSend("jobId", jobId) : {}),
-      ...(source ? SQSUtils.singleAttributeToSend("source", source) : {}),
-    },
+  const messageAtribs = {
+    ...SQSUtils.singleAttributeToSend("cxId", cxId),
+    ...SQSUtils.singleAttributeToSend("patientId", patientId),
+    ...(jobStartedAt ? SQSUtils.singleAttributeToSend("jobStartedAt", jobStartedAt) : {}),
+    ...(jobId ? SQSUtils.singleAttributeToSend("jobId", jobId) : {}),
+    ...(medicalDataSource ? SQSUtils.singleAttributeToSend("source", medicalDataSource) : {}),
   };
-  await sqs.sendMessage(sendParams).promise();
+  await Promise.all([
+    sqs
+      .sendMessage({
+        MessageBody: queuePayload,
+        QueueUrl: fhirServerQueueURL,
+        MessageAttributes: messageAtribs,
+      })
+      .promise(),
+    sqs
+      .sendMessage({
+        MessageBody: queuePayload,
+        QueueUrl: patientDataConsolidatorQueueURL,
+        MessageAttributes: messageAtribs,
+        // This is critical to make sure we only process one message per patient at a time
+        MessageGroupId: toMessageGroupId(patientId),
```

Comment:
Can you explain what the platform implications of this are? wont this slow down the converter processing a lot?

Reply:
and why are we doing this here as opposed to in sqs-to-consolidated?

Reply:
This is not to convert, this is to the consolidator. This is slower, yes, but we need a way to "synchronize" (race control) the updates to the consolidated bundle, to prevent two lambdas loading the consolidated file, updating it in memory w/ its payload, and writing it back - one of the payloads would be lost.

Reply:
The queue from the converter to the consolidator is FIFO ([differences](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-queue-types.html)).

FIFO queues require a message group ID and message ID:
- message group ID defines a virtual queue inside the queue ([ref](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/using-messagegroupid-property.html));
- message deduplicationID defines the unique identifier for the message.

For more details: https://hevodata.com/learn/sqs-fifo-queues/

This allows us to have a single queue between the converter and the consolidator, but make sure that only one message at a time is processed for each patient.

Reply:
ok makes sense. thanks for clarifying